src/app/my-donations/my-donations.component.ts

```diff
@@ -35,7 +35,23 @@ export class MyDonationsComponent implements OnInit{
     );
 
     this.donations = this.route.snapshot.data.donations;
-    this.atLeastOneDonationWasLarge = this.donations.some(isLargeDonation);
+
+
+    this.atLeastOneLargeRecentDonation = this.donations
+      .filter(donationIsRecent)
+      .some(isLargeDonation);
+
+    function donationIsRecent(donation: Donation) {
```

Comment:
this isn't automatically tested, but I have tested it manually, and it's fairly straighforward and not very high impact.